
Adds support for SoC's unique ID (Maverick Key) in /proc/cpuinfo

Signed-off-by: Petr Stetiar <ynezz@true.cz>

Index: linux-2.6.24/arch/arm/kernel/setup.c
===================================================================
--- linux-2.6.24.orig/arch/arm/kernel/setup.c	2010-02-07 14:53:58.000000000 +0100
+++ linux-2.6.24/arch/arm/kernel/setup.c	2010-02-07 14:54:35.000000000 +0100
@@ -1007,8 +1007,15 @@
 
 	seq_printf(m, "Hardware\t: %s\n", machine_name);
 	seq_printf(m, "Revision\t: %04x\n", system_rev);
+
+#if defined(CONFIG_ARCH_EP93XX)
+#include <asm/arch/ep93xx-regs.h>
+	seq_printf(m, "Serial\t\t: %016x\n",
+		   *((unsigned int *)EP93XX_SECURITY_UNIQID));
+#else
 	seq_printf(m, "Serial\t\t: %08x%08x\n",
 		   system_serial_high, system_serial_low);
+#endif
 
 	return 0;
 }
Index: linux-2.6.24/include/asm-arm/arch-ep93xx/ep93xx-regs.h
===================================================================
--- linux-2.6.24.orig/include/asm-arm/arch-ep93xx/ep93xx-regs.h	2010-02-07 14:54:29.000000000 +0100
+++ linux-2.6.24/include/asm-arm/arch-ep93xx/ep93xx-regs.h	2010-02-07 14:54:35.000000000 +0100
@@ -70,6 +70,8 @@
 #define EP93XX_I2S_BASE			(EP93XX_APB_VIRT_BASE + 0x00020000)
 
 #define EP93XX_SECURITY_BASE		(EP93XX_APB_VIRT_BASE + 0x00030000)
+#define EP93XX_SECURITY_REG(x)		(EP93XX_SECURITY_BASE + (x))
+#define EP93XX_SECURITY_UNIQID		EP93XX_SECURITY_REG(0x2440)
 
 #define EP93XX_GPIO_BASE		(EP93XX_APB_VIRT_BASE + 0x00040000)
 #define EP93XX_GPIO_REG(x)		(EP93XX_GPIO_BASE + (x))
